x-pack/agentbeat/magefile.go

```diff
@@ -216,11 +228,39 @@ func PythonIntegTest(ctx context.Context) error {
 }
 
 func SystemTest(ctx context.Context) error {
-	mg.SerialDeps(xpacketbeat.GetNpcapInstallerFn("../packetbeat"), Update, devtools.BuildSystemTestBinary)
+	if slices.Contains(getIncludedBeats(), "packetbeat") {
+		mg.SerialDeps(xpacketbeat.GetNpcapInstallerFn("../packetbeat"), Update, devtools.BuildSystemTestBinary)
 
-	args := devtools.DefaultGoTestIntegrationArgs()
-	args.Packages = []string{"../packetbeat/tests/system/..."}
-	args.Tags = append(args.Tags, "agentbeat")
+		args := devtools.DefaultGoTestIntegrationArgs()
+		args.Packages = []string{"../packetbeat/tests/system/..."}
+		args.Tags = append(args.Tags, "agentbeat")
+
+		return devtools.GoTest(ctx, args)
+	}
+	return nil
+}
+
+func getIncludedBeats() []string {
+	// beats are all the beats the agentbeat can combine
+	includedBeats := []string{
+		"auditbeat",
+		"filebeat",
+		"heartbeat",
+		"metricbeat",
+		"osquerybeat",
+		"packetbeat",
+	}
+
+	if devtools.FIPSBuild {
+		// If a FIPS-compliant version of agentbeat is being built, restrict the list of beats to the fips-compliant ones
```

Comment:
```suggestion
		// If a FIPS-compliant version of agentbeat is being built, restrict the list of beats to the fips-compliant ones. Filtering allows us to produce a FIPS complaint beat that is not included in agentbeat
```
(nitpick)

Reply:
I am not sure I understand: with this modification we are creating a FIPS compliant agentbeat so 
`Filtering allows us to produce a FIPS complaint beat that is not included in agentbeat` doesn't sound very clear.
What about `Removing non compliant beats helps produce a compliant agentbeat,  excluding code that has not been designed for a FIPS environment`

Reply:
Added an extra comment in bd2e91e666a5f129f4cf1b2d7f86448fc4c731bc

Reply:
👍 I though we should add a comment to explain why we don't just use `devtools.FIPSConfig.Beats` directly